static/js/compose.js

```diff
@@ -24,10 +24,13 @@ function make_uploads_relative(content) {
     return content.replace(exports.uploads_re, "]($1)");
 }
 
-function show_all_everyone_warnings() {
+function show_all_everyone_warnings(is_announce_stream) {
```

Comment:
With this change, the function name doesn't really reflect the functionality anymore. I'd prefer renaming this function to smt like `show_warnings(warning_type)`, where `warning_type` is a value in {'all', 'everyone', 'announce'}, and then change the function logic and calls respectively.